File: solutions/1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.cs
public class Solution {
    public int MaxEqualRowsAfterFlips(int[][] matrix) {
        var count = new Dictionary<string, int>();
        
        foreach (var row in matrix) {
            var key = new int[row.Length];
            for (int i = 0; i < row.Length; i++) {
                key[i] = row[0] == 1 ? 1 - row[i] : row[i];
            }
            var rowKey = string.Join(",", key);
            count[rowKey] = count.GetValueOrDefault(rowKey, 0) + 1;
        }
        
        return count.Values.Max();
    }
}

File: solutions/1147-flip-columns-for-maximum-number-of-equal-rows/flip-columns-for-maximum-number-of-equal-rows.ts
// TypeScript

function maxEqualRowsAfterFlips(matrix: number[][]): number {
    const count = new Map<string, number>();
    
    for (const row of matrix) {
        const key = row.map(n => row[0] ? 1 - n : n).join(',');
        count.set(key, (count.get(key) || 0) + 1);
    }
    
    return Math.max(...count.values());
}